src/frontend/src/catalog/table_catalog.rs

```diff
@@ -570,7 +574,13 @@ impl From<PbTable> for TableCatalog {
 
         let conflict_behavior = ConflictBehavior::from_protobuf(&tb_conflict_behavior);
         let version_column_index = tb.version_column_index.map(|value| value as usize);
-        let columns: Vec<ColumnCatalog> = tb.columns.into_iter().map(ColumnCatalog::from).collect();
+        let mut columns: Vec<ColumnCatalog> =
+            tb.columns.into_iter().map(ColumnCatalog::from).collect();
+        let rw_timestamp_column = ColumnCatalog::rw_timestamp_column();
+        if !columns.contains(&rw_timestamp_column) {
+            // Add system column `_rw_timestamp` to every table, but notice that this column is never persisted.
+            columns.push(rw_timestamp_column);
+        }
```

Comment:
This looks a bit hack to me. Since it's never persisted, can we use a separate fields to avoid confusion with real persisted columns?

Reply:
I don't want to persist this column because we have many already persisted tables in prod. Users won't want to rebuild their tables to use this feature.

Reply:
Yes, I agree with that. I'm just wondering if it's possible to not have this column in runtime `TableCatalog` at all, as all other columns are persisted.

Reply:
I admit it is a bit hacky. If you take a look at how `LogicalScan` constructs the schema, it uses table catalog columns directly. If we want to keep the table catalog the same as before, we need to add complexity to the table scan and change how it represents the schema, and I think `TableScan` schema construction is already quite complex. So how about adding an additional field to `ColumnDesc` to distinguish the system column from others?